sdk/include/asm/asm.inc

```diff
@@ -100,7 +100,7 @@ ENDM
 
 .code16 MACRO
     ASSUME nothing
-    .text SEGMENT use16
+    .text SEGMENT use16 PUBLIC 'CODE'
```

Comment:
That's the magic so that, when compiling and linking freeldr, we don't get *two separate* .text segments with different attributes, *even if* we ask all the .text/.data ... segments to be merged together in the freeldr linker command-line.

This avoids the `"v86.S.obj : warning LNK4078: multiple '.text' sections found with different attributes (C0520040)"` warning.

I have checked the generated code (freeldr_pe.exe) in IDA to see whether the incorporated 16-bit code was OK and it seemed to be so.

Reply:
Maybe insert this as a comment?

Reply:
Nice! This might help to remove the freeldr16.bin hack at some point.